executor/aggfuncs/builder.go

```diff
@@ -135,6 +137,42 @@ func buildApproxCountDistinct(aggFuncDesc *aggregation.AggFuncDesc, ordinal int)
 	return nil
 }
 
+func buildApproxPercentile(sctx sessionctx.Context, aggFuncDesc *aggregation.AggFuncDesc, ordinal int) AggFunc {
+	if aggFuncDesc.Mode == aggregation.DedupMode {
+		return nil
+	}
+
+	// Checked while building descriptor
+	percent, _, err := expression.GetIntFromConstant(sctx, aggFuncDesc.Args[1])
+	if err != nil {
+		// Should not reach here
+		panic(fmt.Sprintf("Error happened when buildApproxPercentile: %v", err))
```

Comment:
This panic won't happen since it has been checked at:
https://github.com/pingcap/tidb/blob/8627dc545a18e0328d8d8a68023bdd6bb058a732/expression/aggregation/base_func.go#L143-L152

This check is only for ci check_dev, just as `buildGroupConcat` did:
https://github.com/pingcap/tidb/blob/8627dc545a18e0328d8d8a68023bdd6bb058a732/executor/aggfuncs/builder.go#L437-L449

Reply:
ignore the error?
percent, _, _ := expression.GetIntFromConstant(sctx, aggFuncDesc.Args[1])

Reply:
If we do so, it will be blocked by `idc-jenkins-ci-tidb/check_dev_2` or `idc-jenkins-ci-tidb/check_dev`, since there is an unchecked error.

Reply:
Maybe we could log the error here.

Reply:
Log it please.

Reply:
@fzhedu Fixed. Then do we need to panic it?